core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java

```diff
@@ -731,9 +731,22 @@ public Map<String, Boolean> updateConfiguration(Map<String, String> propertiesMa
         PropertyKey key = PropertyKey.fromString(entry.getKey());
         if (Configuration.getBoolean(PropertyKey.CONF_DYNAMIC_UPDATE_ENABLED)
             && key.isDynamic()) {
-          Object oldValue = Configuration.get(key);
-          Object value = key.parseValue(entry.getValue());
-          Configuration.set(key, value, Source.RUNTIME);
+          Object oldValue = null;
+          try {
+            oldValue = Configuration.get(key);
+          } catch (RuntimeException e) {
+            LOG.debug("Cannot get old value for the given key {}", key.getName());
+          }
+          if ("<UNSET_VALUE>".equals(entry.getValue())) {
+            Configuration.unset(key);
```

Comment:
this unset does not set it back to default?

Reply:
the unset test case can verify it, also, I double check the unset api, it just call the `AlluxioProperties#remove`, which can affect only while the key is in the `mUserProps`

```java
  public void remove(PropertyKey key) {
    // remove is a nop if the key doesn't already exist
    if (mUserProps.containsKey(key)) {
      mUserProps.remove(key);
      mSources.remove(key);
      mHash.markOutdated();
    }
  }
```